doc/microsite/src/main/tut/distage/distage-testkit.md

```diff
@@ -3,10 +3,265 @@ distage-testkit
+2. Override the `config` method to customize the @scaladoc[`TestConfig`](izumi.distage.testkit.TestConfig)
+3. Establish test case context using [`should`](https://www.scalatest.org/scaladoc/3.2.0/org/scalatest/verbs/ShouldVerb.html),
+   [`must`](https://www.scalatest.org/scaladoc/3.2.0/org/scalatest/verbs/MustVerb.html),
+   and [`can`](https://www.scalatest.org/scaladoc/3.2.0/org/scalatest/verbs/CanVerb.html).
+4. Introduce test cases using one of the `in` methods. Test cases can have a variety of forms. From
+   pure functions returning an
+   [assertion](https://www.scalatest.org/scaladoc/3.2.0/org/scalatest/Assertions.html) to effectful
+   functions with dependencies.
+  - No effect type / `Identity` - @scaladoc[`in`](izumi.distage.testkit.services.scalatest.dstest.DistageAbstractScalatestSpec$$LowPriorityIdentityOverloads)
+  - @scaladoc[`in` for `F[_]`](izumi.distage.testkit.services.scalatest.dstest.DistageAbstractScalatestSpec$$DSWordSpecStringWrapper)
+  - @scaladoc[`in` for `F[+_, +_]`](izumi.distage.testkit.services.scalatest.dstest.DistageAbstractScalatestSpec$$DSWordSpecStringWrapper2)
+  - @scaladoc[`in` for `F[-_, +_, +_]`](izumi.distage.testkit.services.scalatest.dstest.DistageAbstractScalatestSpec$$DSWordSpecStringWrapper3)
+  - Magnet for test cases dependent on injectables: @scaladoc[`ProviderMagnet`](izumi.distage.model.providers.ProviderMagnet)
+
+### API Overview
+
+To demonstrate the usage of `distage-testkit` we'll consider a game score service. This will have a
+model, logic and service which we'll then define test cases to verify.
+
+The highest value tests to develop, [in our experience](https://www.youtube.com/watch?v=CzpvjkUukAs),
+are those that verify communication behavior of components. These are tests of blackbox interfaces,
+with atomic or group isolation levels.
+
+(TODO: expand on dual test tactic here)
+
+For demonstration, we'll create a hypothetical application in `ZIO[-R, +E, +A]`. Starting with a
+model and service:
+
+```scala mdoc:to-string
+import izumi.distage.model.definition.ModuleDef
+import zio._
+
+case class Score(value: Int)
+
+case class Config(starValue: Int, mangoValue: Int)
+
+trait BonusService {
+  def queryCurrentBonus: UIO[Int]
+}
+
+object Score {
+  def addStar(config: Config, score: Score) =
+    score.copy(value = score.value + config.starValue)
+
+  def echoConfig(config: Config): Task[Config] = Task {
+    println(config.toString)
+    config
+  }
+
+  def addMango(config: Config, score: Score): RIO[Has[BonusService], Score] =
+    for {
+        bonusService <- RIO.service[BonusService]
+        currentBonus <- bonusService.queryCurrentBonus
+    } yield {
+        val value = score.value + config.mangoValue + currentBonus
+        score.copy(value = value)
+    }
+
+  def addMango(score: Score): RIO[Has[BonusService] with Has[Config], Score] =
+    RIO.service[Config] flatMap { config => addMango(config, score) }
+}
+
+```
+
+#### `SpecScalatest` Base Classes
```

Comment:
```suggestion
#### `DistageSpecScalatest` Base Classes
```
I suppose it's a bit wierd to omit the prefix (even though there's a clash with the actual class name)